src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -3612,27 +3547,20 @@ public override SyntaxNode ObjectCreationExpression(SyntaxNode type, IEnumerable
             return SyntaxFactory.ObjectCreationExpression((TypeSyntax)type, CreateArgumentList(arguments), null);
         }
 
-        private ArgumentListSyntax CreateArgumentList(IEnumerable<SyntaxNode> arguments)
+        private static ArgumentListSyntax CreateArgumentList(IEnumerable<SyntaxNode> arguments)
         {
             return SyntaxFactory.ArgumentList(CreateArguments(arguments));
         }
 
-        private SeparatedSyntaxList<ArgumentSyntax> CreateArguments(IEnumerable<SyntaxNode> arguments)
+        private static SeparatedSyntaxList<ArgumentSyntax> CreateArguments(IEnumerable<SyntaxNode> arguments)
         {
-            return SyntaxFactory.SeparatedList(arguments.Select(AsArgument).Cast<ArgumentSyntax>());
+            return SyntaxFactory.SeparatedList(arguments.Select(AsArgument));
```

Comment:
> Also, given #6642 I think we can remove lambdas eventually.

I have an analyzer that suggests doing exactly that.  But i'm not going to check-in/enable until we get that actual cache change in as it can actually lead to worse perf allocation-wise.